### PR TITLE
Functional tests -  add test 'Filter modules by status'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add functional tests **Filter modules by status** in module manager page
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/modules/moduleManager/01_filterModulesByStatus.js" URL_FO=Shop_URL npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17027)

### CODE REVIEW

**Review by SimonGrn — DISMISSED** [review]

File: tests/puppeteer/pages/BO/modules/moduleManager/index.js [hunk 1]
```diff
@@ -38,4 +44,50 @@ module.exports = class moduleManager extends BOBasePage {
...
+    const allModulesNames = await this.getAllModulesNames();
+    for (let i = 0; i < allModulesNames.length; i++) {
+      const moduleStatus = await this.isModuleEnabled();
+      await modulesStatus.push({name: allModulesNames[i], status: moduleStatus});
+    }
+    return modulesStatus;
+  }
+
+  async getAllModulesNames() {
+    return this.page.$$eval(
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/modules/moduleManager/index.js:88]: What is `$$eval` ?

Comment (SimonGrn) [reply tests/puppeteer/pages/BO/modules/moduleManager/index.js:88]: A puppeteer command to let us execute Javascript code in the page itself. It's really useful to target specific stuff or trigger events (like clicks) when Puppeteer is not behaving correctly.
We use it in a lot of places in the tests actually :)

Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/modules/moduleManager/index.js:88]: for more information https://pptr.dev/#?product=Puppeteer&version=v1.9.0&show=api-pageevalselector-pagefunction-args-1

Comment (Progi1984) [reply tests/puppeteer/pages/BO/modules/moduleManager/index.js:88]: Top. Thanks

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**